### ISSUE TITLE
Problem with Http.RequestString detection of response content encoding

### ISSUE THREAD

**Issue by dam5s** [fsharp/FSharp.Data#1251]
When using Http.RequestString to download "https://www.lemonde.fr/rss/une.xml", the resulting string has unexpected character encoding issues:
```
      <title>Le parc locatif privÃ© bousculÃ© par lâessor des locations meublÃ©es</title>
```

instead of the output produced when using System.Net.Http directly:
```
      <title>Le parc locatif privé bousculé par l’essor des locations meublées</title>
```

I created a small sample project to reproduce the issue and compare to System.Net -
https://github.com/dam5s/fsharp-data-requeststring-issue

I'm running it on .NET core 2.2 on Linux.
Let me know if more details are needed, thanks!

**Comment by ssimontis** [comment]
Hey there dam5s,

I think this is an encoding issue. If you dig into the .NET Core source code (I used JetBrains dotPeek to explore .NET Core App v2.2.3 of System.Net.Http, the class *HttpContent* defines the default encoding of a response as UTF-8, which is used if a CharSet is not specified in the HTTP ContentType response header. The XML file does not return a CharSet in the header; it is specified in the XML declaration on the first line of the file as UTF-8.

FSharp.Data.Http.fs defines the default encoding of a response as ISO-8859-1 which is basically UTF-8 forced into 1 byte. If the language has > 127 code points, proper output is not returned, which seems to be what you are experiencing.  The source code has a [comment to this SO post on default encoding of HTTP POST requests_[http://stackoverflow.com/questions/708915/detecting-the-character-encoding-of-an-http-post-request/708942#708942] as justification for this, but it appears UTF-8 is used by most browsers now.

In short, I believe that FSharp.Data and .NET Core disagree upon the default encoding to use with responses. I have never tried to help out on an open source project before, I randomly picked your issue to investigate and this is my finding. I guess I need to find a representative of the project and contact them about a possible fix?

**Comment by dam5s** [comment]
It depends.

It seems the project is following the RFC to the letter, which then breaks websites/servers assuming that clients should use UTF-8 as the default encoding.

It seems browsers, curl and the System.Http implementation disagree with that stance and do default to UTF-8 instead.

Does any maintainer have any input on that?

**Comment by ssimontis** [comment]
@ovatsus @tpetricek @colinbull Could you please take a look at this and advise on which behavior this project wishes to implement? Right now, it is following the RFC for POST encoding, but this puts it at odds with System.Net.Http's behavior. I am happy to fix the issue, I just need to know what you think is best for the project. Thanks!

**Comment by colinbull** [comment]
Ok, so my two cents is that there isn’t really a “correct” way to do this looking at the findings above. To this end I propose one of two things. Add a ‘http.setfallbackencoding’ function to the pipeline and let the user specify the encoding to use when the CharSet header is not specified. Or add an overload to request string that allows you to pass the default encoding as a parameter when CharSet is not specified. In terms of API design I think the latter is probably more in style with this lib.

**Comment by ssimontis** [comment]
@colinbull Thanks for getting back to us so quickly! I will take a look at this after work tonight.

**Comment by colinbull** [comment]
No worries, if you need any help just let me know.

**Comment by suside** [comment]
According to [RFC 4627](http://www.ietf.org/rfc/rfc4627.txt) UTF-8 is the default encoding for JSON. 
So it looks like `HttpEncodings.ResponseDefaultEncoding` should be split between:

1. text, RFC2616, ISO-8859-1
2. json, RFC4627, UTF-8

However `System.Net.Http.HttpClient` is using UTF-8 even for `Content-type: text/plain`

Workaround for now is to use `Http.RequestString(..., responseEncodingOverride = "utf-8")`
